core/src/main/java/io/questdb/cairo/sql/async/PageFrameReduceJob.java

```diff
@@ -186,6 +179,10 @@ private static boolean consumeQueue(
                             .$(", cursor=").$(cursor)
                             .I$();
                     if (frameSequence.isActive()) {
+                        if (workerId > -1) {
```

Comment:
This check may be insufficient because of work stealing. In case of work stealing, the `workerId` is set to `-1`, but the frame sequence may be different. In that case, we want to behave as if it was a worker thread running a `PageFrameReduceJob`. So, the check should be something like `workerId != -1 || frameSequence != stealingFrameSequence`. Also, it may be worth it expanding the clarifying comment around the check.

Reply:
good catch, fixing it.

Reply:
done